src/main/java/uk/gov/pay/api/validation/ExactLengthOrEmpty.java

```diff
@@ -0,0 +1,29 @@
+package uk.gov.pay.api.validation;
+
+import javax.validation.Constraint;
+import javax.validation.Payload;
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE_USE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+@Target({ FIELD, METHOD, PARAMETER, ANNOTATION_TYPE, TYPE_USE })
+@Retention(RUNTIME)
+@Constraint(validatedBy = ExactLengthOrEmptyValidator.class)
+@Documented
+public @interface ExactLengthOrEmpty {
+
+    String message() default "Must be exactly 2 characters length";
+
+    Class<?>[] groups() default {};
+
+    Class<? extends Payload>[] payload() default {};
+
+    int size();
```

Comment:
Needs rebasing but just a quick comment, but might be nice to call this `length` so its consistent with the wording of the annotation and the attribute in `ExactLengthOrEmptyValidator` which is `length`. Not a bit problem so don't go out of your way to do it if.

Reply:
i've fixed this